tests/datatests/test_new_tasks.py

```diff
@@ -59,6 +60,14 @@ def test_verify_data(self):
                 parser = setup_args()
                 opt = parser.parse_args(args=['--task', subt], print_args=False)
                 opt['task'] = subt
+                if subt.split(':')[-1] == 'ParlaiformatTeacher':
```

Comment:
wait actually now i am confused; why does your test fail if you do not include this block?

Reply:
Because the `fromfile` agent requires `--fromfile-datapath` to be set: https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/fromfile/agents.py#L81

Reply:
but don't you set it before you call `super().__init__(opt, shared)`?

Reply:
Hmm, this flag isn't set at all if you just call `fromfile` by itself. You can see this by calling `python examples/display_data.py -t fromfile`:
```
Traceback (most recent call last):
  File "examples/display_data.py", line 19, in <module>
    DisplayData.main()
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/scripts/script.py", line 79, in main
    return cls._run_args(None)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/scripts/script.py", line 66, in _run_args
    return script.run()
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/scripts/display_data.py", line 108, in run
    return display_data(self.opt)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/scripts/display_data.py", line 71, in display_data
    world = create_task(opt, agent)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/core/worlds.py", line 1627, in create_task
    world = create_task_world(opt, user_agents, default_world=default_world)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/core/worlds.py", line 1590, in create_task_world
    task_agents = _create_task_agents(opt)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/core/worlds.py", line 1578, in _create_task_agents
    return create_task_agent_from_taskname(opt)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/core/teachers.py", line 2401, in create_task_agent_from_taskname
    task_agents = teacher_class(opt)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/tasks/fromfile/agents.py", line 114, in __init__
    super().__init__(opt, shared)
  File "/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/tasks/fromfile/agents.py", line 82, in __init__
    raise RuntimeError('fromfile_datapath not specified')
RuntimeError: fromfile_datapath not specified
```

The interesting thing about `test_new_tasks.py` is that it'll test any teachers imported *alongside* a module even if they're not defined in the module. This happens as a result of this line: https://github.com/facebookresearch/ParlAI/blob/b9e1027a72e7baad0d21fce00a0d58c66e3c3fef/tests/datatests/test_new_tasks.py#L45 This is why I have to get this test working for `-t fromfile` even though my PR doesn't change the `fromfile` teacher

Reply:
So given that your teachers are `ParlaiFormat`, would it be possible to just have them subclass `ParlAIDialogTeacher` instead?

Reply:
Hmm, okay, I'll try that, but I feel like if we remove the code I just added to handle `fromfile`, we'll hit this issue again whenever as someone else subclasses fromfile

Reply:
Yeah i guess this begs a bigger question of whether we want people to subclass `fromfile` when you get equivalent functionality by subclassing `ParlAIDialogTeacher`. I think `fromfile` is meant for ease of use on the command-line / quick dataset viewing and manipulation. As a real teacher, fromfile as u see just wraps these actual teachers